src/internal/kinesis.ts

```diff
@@ -0,0 +1,490 @@
+/**
+ * It creates a new KinesisClient object and returns it.
+ * @param {T} obj - T - The object that we're checking for the key.
+ * @param k - The number of virtual users to simulate.
+ */
+import { JSONArray, JSONObject } from 'k6'
+import http, { RefinedResponse, ResponseType } from 'k6/http'
+
+import { AWSClient } from './client'
+
+import { AWSConfig } from './config'
+import { AMZ_TARGET_HEADER } from './constants'
+import { AWSError } from './error'
+import { HTTPHeaders } from './http'
+import { InvalidSignatureError, SignatureV4 } from './signature'
+
+
+/**
+This API is based on
+https://docs.aws.amazon.com/kinesis/latest/APIReference/API_Operations.html
+*/
+
+
+/* This class is a wrapper around the AWS Kinesis client. It provides a method for each Kinesis
+operation, and each method returns a promise. */
+export class KinesisClient extends AWSClient {
+
+    signature: SignatureV4
+
+    commonHeaders: HTTPHeaders
+
+    serviceVersion: string
+
+    /**
+     * A constructor function that creates a new instance of the Kinesis class.
+     * @param {AWSConfig} awsConfig - AWSConfig - This is the configuration object that is passed to
+     * the constructor.
+     * @returns A Proxy object.
+     */
+    constructor(awsConfig: AWSConfig) {
+        super(awsConfig, 'kinesis')
+
+        this.serviceVersion = 'Kinesis_20131202'
+
+        this.signature = new SignatureV4({
+            service: this.serviceName,
+            region: this.awsConfig.region,
+            credentials: {
+                accessKeyId: this.awsConfig.accessKeyId,
+                secretAccessKey: this.awsConfig.secretAccessKey,
+                sessionToken: this.awsConfig.sessionToken,
+            },
+            uriEscapePath: false,
+            applyChecksum: true,
+        })
+
+
+        this.commonHeaders = {
+            'Content-Type': 'application/x-amz-json-1.1',
+        }
+
+        return new Proxy(this, {
```

Comment:
I would not like to limit the ability of this library, until all functions are available. 
I hope this fine for functionality as concrete functions are already implemented.

Reply:
I understand where you come from @autometa101. However, as mentioned in a previous iteration, we would like to avoid having "magic" code (as we need to document the available API to users in k6's documentation), and we would also like to avoid code meant for a future use case.

Let's, please, drop that call for now 🙇🏻